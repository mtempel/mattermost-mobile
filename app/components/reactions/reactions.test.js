// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Reactions from './reactions';

import Preferences from 'mattermost-redux/constants/preferences';

describe('Reactions', () => {
    const baseProps = {
        actions: {
            addReaction: jest.fn(),
            getReactionsForPost: jest.fn(),
            removeReaction: jest.fn(),
        },
        canAddReaction: true,
        canAddMoreReactions: true,
        canRemoveReaction: true,
        currentUserId: 'current-user-id',
        position: 'right',
        postId: 'post-id',
        reactions: {
            'current-user-id-frowning_face': {create_at: 1, emoji_name: 'frowning_face', post_id: 'post-id', user_id: 'current-user-id'},
            'current-user-id-grinning': {create_at: 1, emoji_name: 'grinning', post_id: 'post-id', user_id: 'current-user-id'},
            'current-user-id-sweat': {create_at: 1, emoji_name: 'sweat', post_id: 'post-id', user_id: 'current-user-id'},
        },
        theme: Preferences.THEMES.default,

    };

    test('should match snapshot', () => {
        const wrapper = shallow(<Reactions {...baseProps}/>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot with canAddReaction = false', () => {
        const wrapper = shallow(
            <Reactions
                {...baseProps}
                canAddReaction={false}
            />,
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
