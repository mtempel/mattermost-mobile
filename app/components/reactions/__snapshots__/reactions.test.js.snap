// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reactions should match snapshot 1`] = `
<View
  style={
    Object {
      "alignContent": "flex-start",
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
>
  <Reaction
    count={1}
    emojiName="frowning_face"
    highlight={true}
    onLongPress={[Function]}
    onPress={[Function]}
    postId="post-id"
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBg": "#ffffff",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
  <Reaction
    count={1}
    emojiName="grinning"
    highlight={true}
    onLongPress={[Function]}
    onPress={[Function]}
    postId="post-id"
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBg": "#ffffff",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
  <Reaction
    count={1}
    emojiName="sweat"
    highlight={true}
    onLongPress={[Function]}
    onPress={[Function]}
    postId="post-id"
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBg": "#ffffff",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
  <TouchableWithFeedbackIOS
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "rgba(61,60,64,0.3)",
          "borderRadius": 2,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginRight": 6,
          "marginTop": 10,
          "paddingHorizontal": 6,
          "paddingVertical": 2,
          "width": 40,
        },
      ]
    }
    type="opacity"
  >
    <Image
      source={
        Object {
          "testUri": "../../../dist/assets/images/icons/reaction.png",
        }
      }
      style={
        Object {
          "height": 20,
          "tintColor": "rgba(61,60,64,0.5)",
          "width": 23,
        }
      }
    />
  </TouchableWithFeedbackIOS>
</View>
`;

exports[`Reactions should match snapshot with canAddReaction = false 1`] = `
<View
  style={
    Object {
      "alignContent": "flex-start",
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
>
  <Reaction
    count={1}
    emojiName="frowning_face"
    highlight={true}
    onLongPress={[Function]}
    onPress={[Function]}
    postId="post-id"
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBg": "#ffffff",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
  <Reaction
    count={1}
    emojiName="grinning"
    highlight={true}
    onLongPress={[Function]}
    onPress={[Function]}
    postId="post-id"
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBg": "#ffffff",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
  <Reaction
    count={1}
    emojiName="sweat"
    highlight={true}
    onLongPress={[Function]}
    onPress={[Function]}
    postId="post-id"
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBg": "#ffffff",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
</View>
`;
